view/adminhtml/web/vue/MenuType.vue

```diff
@@ -0,0 +1,56 @@
+<template>
+    <fieldset class="admin__fieldset fieldset-wide">
+        <simple-field
+            :label="config.translation.nodeName"
+            id="node_name"
+            type="textarea"
+            v-model="item.title"
+        >
+        </simple-field>
+        <simple-field
+            :label="config.translation.nodeClasses"
+            id="node_classes"
+            type="text"
+            v-model="item.classes"
+        >
+        </simple-field>
+        <div class="admin__field field field-title">
+            <label
+                class="label admin__field-label"
+                for="node_type"
+            >
+                {{ config.translation.nodeType }}
+            </label>
+            <div class="admin__field-control control">
+                <select
+                    class="admin__control-select"
+                    name="node_type"
+                    id="node_type"
+                    :value="item.type"
+                    @change="changeType($event.target.value)"
+                >
+                    <option value="">{{config.translation.selectNodeType}}</option>
+                    <option v-for="(label, key) in config.nodeTypes" :value="key">
+                        {{ label }}
+                    </option>
+                </select>
+            </div>
+        </div>
+        <component :is="item['type']" :item="item" :config="config"></component>
+    </fieldset>
+</template>
+
+<script>
+    define(["Vue"], function(Vue) {
+        Vue.component("snowdog-menu-type", {
+            template: template,
+            props   : ['item', 'config'],
+            methods : {
+                changeType: function(value) {
+                    this.item['type'] = value;
+                    this.item['content'] = null;
```

Comment:
Is there an option to keep the content of node per type, to be able to go back to previous node type without losing the content?
Ofc we are not going to keep it on the backend, but as long as the user doesn't refresh a page, it should be possible.

Reply:
localstorage ?